### ISSUE TITLE
Sending mails with empty user/password (no-auth) not working

### ISSUE THREAD

**Issue by flortsch** [eikek/sharry#243]
Hello,

I just set up Sharry (1.4.2) on my server and so far, I really like it :)
On my server I have a Postfix mail server that allows local machine users to send out mails without authentication.
However, sending mails from Sharry with no authentication does not seem to work when I want to share a link.

My smtp config (with empty user as suggested by default config for no-auth):
```
mail {
    enabled = true
    smtp {
        host = "localhost"
        port = 25
        user = ""
        password = ""
        ssl-type = "none"
        check-certificates = true
        timeout = "10 seconds"
        default-from = "Sharry <no-reply@mydomain.com>"
        list-id = "Sharry"
    }
}
```

Log output:
```
[ioapp-compute-1] DEBUG e.j.i.ConnectionResource$ - Creating session without authenticator and props: {mail.mime.multipart.ignoreexistingboundaryparameter=true, mail.smtp.writetimeout=10000, mail.smtp.port=25, mail.mime.parameters.strict=false, mail.mime.multipart.ignoremissingboundaryparameter=true, mail.smtp.auth=true, mail.smtp.timeout=10000, mail.smtp.host=localhost, mail.smtp.auth.mechanisms=XOAUTH2 LOGIN PLAIN DIGEST-MD5 NTLM, mail.smtp.connectiontimeout=10000}
[ioapp-compute-1] DEBUG s.r.r.MailRoutes - Sending mail: SendFailure(javax.mail.AuthenticationFailedException: failed to connect, no password specified?)
```

Looking at the log, it seems that Sharry still tries to somehow authenticate against the mail server, although the user is set to be empty.

Cheers,
Florian

**Comment by eikek** [comment]
Hi @flortsch , thank you for reporting this! I'm going to take a look. It really seems that despite logging "to create a session without authenticator", it tries to authenticate anyways. Does your postfix server advertise some auth mechanisms (maybe for non-local users)?

**Comment by flortsch** [comment]
Thanks for the reply @eikek. My Postfix does not have any auth enabled, as I use Dovecot submission on port 587 for non-local user relaying. However, it allows users from the local network to relay without any auth.

```
Trying ::1...
Connected to localhost.
Escape character is '^]'.
220 mymailserver.mydomain ESMTP
ehlo localhost
250-mymailserver.mydomain
250-PIPELINING
250-SIZE 10240000
250-ETRN
250-STARTTLS
250-ENHANCEDSTATUSCODES
250-8BITMIME
250-DSN
250-SMTPUTF8
250 CHUNKING
auth
503 5.5.1 Error: authentication not enabled
```

No auth mechanisms reported. So it is strange that in the log `mail.smtp.auth.mechanisms=XOAUTH2 LOGIN PLAIN DIGEST-MD5 NTLM` is reported.

**Comment by eikek** [comment]
Thank you a lot for all the details! Yes, it's strange,  I guess it is probably a wrong configuration of javamail. The line in the log are the properties that sharry sets to configure javamail. But it also does exclude the authenticator. So I first though that if there were some mechanisms advertised, javamail might still go trying. But there are none… I think it shouldn't do this, but it is very very likely that I misconfigured it somewhere :). I'm going to reproduce this to find out.

**Comment by eikek** [comment]
I think I found the cause, but I tested against a java-implementation of SMTP. I'm not very confident that it fixes your problem. Before creating a new release, I wanted to ask if you maybe have the time and energy to test a snapshot package? I can fully understand if that is not possible! No problem, I then just release a 1.4.3 and we look from there. The link would be [here](https://box.daheim.site/app/open/HsqfstT8L9p-mjE84EsBsSg-Cr9tPcxAJLb-XX5dvRxeJEm).
